ghost_data.py

```diff
@@ -102,6 +137,40 @@ def ret_value(v: source.ExprVarT[source.VarNameKind]) -> source.ExprVarT[source.
     return v
 
 
+def mkeq(fn_name: source.FunctionName, ty: source.Type) -> source.ExprT[source.HumanVarName]:
+    return eq(source.ExprFunction(ty, fn_name, arguments=(arg_value(lc),)),
+              source.ExprFunction(ty, fn_name, arguments=(ret_value(lc),)))
+
+
+NR_Notification = source.FunctionName('Notification')
+NR_Unknown = source.FunctionName('Unknown')
+
+Ch_empty_fn: source.ExprT[source.HumanVarName] = source.ExprFunction(
+    Set_Ch, Ch_set_empty, ())
+
+
+def hname(name: str, ty: source.Type) -> source.ExprVarT[source.HumanVarName]:
+    return source.ExprVar(ty, source.HumanVarName(source.HumanVarNameSubject(name), use_guard=False, path=()))
+
+
+def wf_MsgInfo() -> source.ExprT[source.HumanVarName]:
+    label_fn_name = source.FunctionName('label')
+    count_fn_name = source.FunctionName('count')
+    just_val = source.ExprFunction(
+        Maybe_MsgInfo, MsgInfo_Just, (arg_value(lc),))
+    label = source.ExprFunction(source.type_word64, label_fn_name, [just_val])
+    count = source.ExprFunction(source.type_word16, count_fn_name, [just_val])
+
+    label_maxval = source.ExprNum(source.type_word64, 0xffffffffffff)
+    count_maxval = source.ExprNum(source.type_word16, 127)
+
+    # TODO: is this correct?
+    return conjs(
+        sle(label, label_maxval),
+        sle(count, count_maxval)
```

Comment:
`sle` stands for `signed less equals`. I don't think signed comparison is what you mean to do here.